### PR TITLE
CAMEL-12983 camel-netty4-http - Allow direct streaming from big files to file output stream to disk

### PR DESCRIPTION
**Pull request by zhfeng**
https://issues.apache.org/jira/browse/CAMEL-12983

This could be a partial fix of this issue and write the steam into the chunked messages.

### CODE REVIEW

File: components/camel-netty4-http/src/main/java/org/apache/camel/component/netty4/http/ChunkedHttpRequest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import io.netty.handler.codec.http.HttpChunkedInput;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.stream.ChunkedInput;
+import io.netty.handler.stream.ChunkedStream;
+
+import java.io.InputStream;
+
+/**
+ * @author <a href="mailto:zfeng@redhat.com">Zheng Feng</a>
+ */
```
Comment (oscerd) [inline components/camel-netty4-http/src/main/java/org/apache/camel/component/netty4/http/ChunkedHttpRequest.java]: Please remove author

File: components/camel-netty4-http/src/main/java/org/apache/camel/component/netty4/http/ChunkedHttpResponse.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpChunkedInput;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.stream.ChunkedInput;
+import io.netty.handler.stream.ChunkedStream;
+
+import java.io.InputStream;
+
+/**
+ * @author <a href="mailto:zfeng@redhat.com">Zheng Feng</a>
```
Comment (oscerd) [inline components/camel-netty4-http/src/main/java/org/apache/camel/component/netty4/http/ChunkedHttpResponse.java]: Please remove author

File: components/camel-netty4-http/src/main/java/org/apache/camel/component/netty4/http/CustomChunkedWriteHandler.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+package org.apache.camel.component.netty4.http;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.stream.ChunkedWriteHandler;
+
+/**
+ * @author <a href="mailto:zfeng@redhat.com">Zheng Feng</a>
```
Comment (oscerd) [inline components/camel-netty4-http/src/main/java/org/apache/camel/component/netty4/http/CustomChunkedWriteHandler.java]: Please remove author

**Comment by zhfeng** [comment]
@oscerd OK, I remove author.

**Comment by oscerd** [comment]
Thanks, merged on master

**Pull request closed without merging**